docs/sidecar/configuration.md

```diff
@@ -0,0 +1,411 @@
+  value: "https://<your-keyvault>.vault.azure.net"
+- name: AzureAd__ClientCredentials__0__KeyVaultCertificateName
+  value: "<certificate-name>"
+```
+
+#### Certificate from File
+
+```yaml
+- name: AzureAd__ClientCredentials__0__SourceType
+  value: "Path"
+- name: AzureAd__ClientCredentials__0__CertificateDiskPath
+  value: "/path/to/certificate.pfx"
+- name: AzureAd__ClientCredentials__0__CertificatePassword
+  value: "<certificate-password>"  # Optional
+```
+
+#### Certificate from Store
+
+```yaml
+- name: AzureAd__ClientCredentials__0__SourceType
+  value: "StoreWithThumbprint"
+- name: AzureAd__ClientCredentials__0__CertificateStorePath
+  value: "CurrentUser/My"
+- name: AzureAd__ClientCredentials__0__CertificateThumbprint
+  value: "<thumbprint>"
+```
+
+#### Signed Assertion
+
+```yaml
+- name: AzureAd__ClientCredentials__0__SourceType
+  value: "SignedAssertionFromManagedIdentity"
+- name: AzureAd__ClientCredentials__0__ManagedIdentityClientId
+  value: "<managed-identity-client-id>"
+```
+
+#### Workload Identity (Recommended for AKS)
+
+```yaml
+# No credentials configuration needed
+# Uses Azure AD Workload Identity via service account
+```
+
+### Credentials Priority
+
+When multiple credentials are configured (indexed 0, 1, 2, etc.), the sidecar attempts them in order until one succeeds:
+
+```yaml
+# First priority - Key Vault certificate
+- name: AzureAd__ClientCredentials__0__SourceType
+  value: "KeyVault"
+- name: AzureAd__ClientCredentials__0__KeyVaultUrl
+  value: "https://prod-keyvault.vault.azure.net"
+- name: AzureAd__ClientCredentials__0__KeyVaultCertificateName
+  value: "prod-cert"
+
+# Second priority - Client secret (fallback)
+- name: AzureAd__ClientCredentials__1__SourceType
+  value: "ClientSecret"
+- name: AzureAd__ClientCredentials__1__ClientSecret
+  valueFrom:
+    secretKeyRef:
+      name: app-secrets
+      key: client-secret
+```
+
+### Downstream APIs Configuration
+
+Configure APIs that your application will call:
+
+```yaml
+- name: DownstreamApis__Graph__BaseUrl
+  value: "https://graph.microsoft.com/v1.0"
+- name: DownstreamApis__Graph__Scopes
+  value: "User.Read Mail.Read"
+- name: DownstreamApis__Graph__RelativePath
+  value: "/me"
+
+- name: DownstreamApis__MyApi__BaseUrl
+  value: "https://api.contoso.com"
+- name: DownstreamApis__MyApi__Scopes
+  value: "api://myapi/.default"
+```
+
+**Per-API Configuration:**
+
+| Key Pattern | Description | Required |
+|-------------|-------------|----------|
+| `DownstreamApis__<Name>__BaseUrl` | Base URL of the API | Yes |
+| `DownstreamApis__<Name>__Scopes` | Space-separated scopes to request | Yes |
+| `DownstreamApis__<Name>__HttpMethod` | Default HTTP method | No (GET) |
+| `DownstreamApis__<Name>__RelativePath` | Default relative path | No |
+| `DownstreamApis__<Name>__RequestAppToken` | Use app token instead of OBO | No (false) |
+
+### Token Acquisition Options
+
+Fine-tune token acquisition behavior:
+
```

Comment:
Here we suppose Graph is the name of one of the downstream API sections you want to call.